AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/editor_component/physx_collider.py

```diff
@@ -0,0 +1,318 @@
+        """
+        Property Type, Default Visibility - ('PhysicsAssetShapeConfiguration', 'ShowChildrenOnly')
+
+        Used to set the Physx Collider's Shape property to PhysicsAsset.
+        """
+        self.component.set_component_property_value(self.Path.SHAPE, physics.ShapeType_PhysicsAsset)
+
+    def set_physx_mesh_from_path(self, asset_product_path: str) -> None:
+        """
+        Property Type, Default Visibility - ('Asset<MeshAsset>', 'Visible')
+
+        Used to set the Physx Collider's PhysicsAsset Shape's Physx Mesh from a user provided asset product path.
+            This will return an error if a source path is provided.
+        """
+        assert self.component.is_property_visible(self.Path.PhysicsAsset.PHYSX_MESH), \
+            f"Failure: Cannot set Physics Mesh when property is not visible."
+
+        px_asset = Asset.find_asset_by_path(asset_product_path)
+        self.component.set_component_property_value(self.Path.PhysicsAsset.PHYSX_MESH, px_asset.id)
+
+    def set_physx_mesh_asset_scale(self, x: float, y: float, z: float) -> None:
+        """
+        Property Type, Default Visibility - ('Vector3', 'Visible')
+
+        Used to set the Physx Collider's Physx Mesh Asset Scale.
+        """
+        assert self.component.is_property_visible(self.Path.PhysicsAsset.ASSET_SCALE), \
+            f"Failure: Cannot set Physics Mesh Asset Scale when property is not visible."
+
+        self.component.set_component_property_value(self.Path.PhysicsAsset.ASSET_SCALE,
+                                                    math.Vector3(float(x), float(y), float(z)))
+
+    def toggle_physics_materials_from_asset(self) -> None:
+        """
+        Property Type, Default Visibility - ('bool', 'Visible')}
+
+        Used to toggle the Physx Collider's Physics Materials From Asset. This allows the user to toggle between
+            using the asset provided PhysX Materials, or to override them using the PhysX Collider's PhysX Materials
+            property.
+        """
+        assert self.component.is_property_visible(self.Path.PhysicsAsset.PHYSICS_MATERIALS_FROM_ASSET), \
+            f"Failure: Cannot toggle Physics Materials From Asset when property is not visible."
+
+        self.component.toggle_property_switch(self.Path.PhysicsAsset.PHYSICS_MATERIALS_FROM_ASSET)
+
+    # Shape: Sphere
+    def set_sphere_shape(self) -> None:
+        """
+        Property Type, Default Visibility - ('SphereShapeConfiguration', 'NotVisible')
+
+        Used to set the Physx Collider's Shape property to Sphere.
+        """
+        self.component.set_component_property_value(self.Path.SHAPE, physics.ShapeType_Sphere)
+
+    def set_sphere_radius(self, radius: float) -> None:
+        """
+        Property Type, Default Visibility - ('float', 'Visible')
+
+        Used to set the Physx Collider's Sphere radius property.
+        """
+        self.component.set_component_property_value(self.Path.Sphere.RADIUS, radius)
```

Comment:
To be consistent with the other modes' property setters, this should check first that `self.Path.Sphere.RADIUS` property is visible and assert if it's not.

Reply:
Updated